terraform/infrastructure/azure/modules/scale_set/main.tf

```diff
@@ -46,9 +46,10 @@ resource "azurerm_linux_virtual_machine_scale_set" "scale_set" {
   disable_password_authentication = false
   upgrade_mode                    = "Manual"
   secure_boot_enabled             = var.secure_boot
-  source_image_id                 = var.image_id
-  tags                            = local.tags
-  zones                           = var.zones
+  # specify the image id only if a non-marketplace image is used
+  source_image_id = var.marketplace_image != null ? null : var.image_id
```

Comment:
Does that mean I can omit the version string in the Constellation config if I set the marketplace image, or is it checked as a required field ahead of time?
I just want to make sure we have a single source of truth for all values and there is no ambiguity.

Reply:
No. If you want to use the marketplace image, we can't use the `source_image_id` attribute, but the `source_image_reference`:

https://github.com/edgelesssys/constellation/blob/2f8ec23d60534d6afe11fb481e5cf10a1ab64027/terraform/infrastructure/azure/modules/scale_set/main.tf#L91

The `var.marketplace_image.version` variable is set to the version of `.image` in `constellation-conf.yaml`,

https://github.com/edgelesssys/constellation/blob/2f8ec23d60534d6afe11fb481e5cf10a1ab64027/cli/internal/cloudcmd/tfvars.go#L172

i.e. the image version field, that would otherwise be resolved to a CSP-specific reference and passed into `source_image_id`.

Reply:
Can you give a quick example of how a config would look like if I want to use the Marketplace with Constellation v2.13.0?

Reply:
sure! It would look like:

```yaml
version: v4 # Schema version of this configuration file.
image: v2.13.0 # Machine image version used to create Constellation nodes.
...
provider:
  azure:
   ...
    useMarketplaceImage: true # Use the Azure Marketplace image instead of a custom image.
...
```